tests/js/client/communication/test-chaos-cluster.js

```diff
@@ -0,0 +1,304 @@
+              c.truncate();
+            } else if (d >= 0.9) {
+              let o = opts();
+              let d = docs();
+              console.warn("RUNNING AQL INSERT WITH " + d.length + " DOCS. OPTIONS: " + JSON.stringify(o));
+              query.call(ctx, "FOR doc IN @docs INSERT doc INTO " + c.name(), {docs: d}, o);
+            } else if (d >= 0.8) {
+              let o = opts();
+              console.warn("RUNNING AQL REMOVE. OPTIONS: " + JSON.stringify(o));
+              query.call(ctx, "FOR doc IN " + c.name() + " LIMIT 138 REMOVE doc IN " + c.name(), {}, o);
+            } else if (d >= 0.7) {
+              let o = opts();
+              console.warn("RUNNING AQL REPLACE. OPTIONS: " + JSON.stringify(o));
+              query.call(ctx, "FOR doc IN " + c.name() + " LIMIT 1338 REPLACE doc WITH { pfihg: 434, fjgjg: 555 } IN " + c.name(), {}, o);
+            } else if (d >= 0.25) {
+              let o = opts();
+              let d = docs();
+              console.warn("RUNNING INSERT WITH " + d.length + " DOCS. OPTIONS: " + JSON.stringify(o));
+              c.insert(d, o);
+            } else {
+              let d = docs();
+              console.warn("RUNNING REMOVE WITH " + d.length + " DOCS");
+              c.remove(d);
+            }
+          } catch (err) {}
+        }
+        
+        if (trx) {
+          if (Math.random() < 0.2) {
+            trx.abort();
+          } else {
+            trx.commit();
+          }
+        }
+      };
+
+      testOpts.collection = cn;
+      code = `(${code.toString()})(${JSON.stringify(testOpts)});`;
+      
+      let tests = [];
+      for (let i = 0; i < 3; ++i) {
+        tests.push(["p" + i, code]);
+      }
+
+      // run the suite for 5 minutes
+      runParallelArangoshTests(tests, 5 * 60, cn);
+
+      print("Finished load test - waiting for cluster to get in sync before checking consistency.");
+      clearAllFailurePoints();
+      waitForShardsInSync(cn);
+      checkCollectionConsistency(cn);
+    }
+  };
+}
+
+function BuildChaosSuite(opts, suffix) {
+  let suite = {};
+  deriveTestSuite(BaseChaosSuite(opts), suite, suffix);
+  return suite;
+}
+
+const params = ["IntermediateCommits", "FailurePoints", "Truncate", "VaryingOverwriteMode", "StreamingTransactions"];
+
+function addSuite(paramValues) {
+  let suffix = "";
+  let opts = {};
+  for (let j = 0; j < params.length; ++j) {
+    suffix += paramValues[j] ? "_with_" : "_no_";
+    suffix += params[j];
+    opts["with" + params[j]] = paramValues[j];
+  }  
+  let func = function() { return BuildChaosSuite(opts, suffix); };
+  // define the function name as it shows up as suiteName
+  Object.defineProperty(func, 'name', {value: "ChaosSuite" + suffix, writable: false});
+
+  jsunity.run(func);
+}
+/*
+for (let i = 0; i < (1 << params.length); ++i) {
+  let paramValues = [];
+  for (let j = params.length - 1; j >= 0; --j) {
+    paramValues.push(Boolean(i & (1 << j)));
+  }
+  addSuite(paramValues);
+}
+*/
```

Comment:
```suggestion
```

Reply:
This code dynamically creates test suites for all parameter combinations. ATM we don't use it since we don't want to include ALL suites in the PR tests, but we do want to have them in the nightlies, but that will be done in a separate PR.